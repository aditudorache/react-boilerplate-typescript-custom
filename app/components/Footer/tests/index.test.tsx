import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';

import Footer from '../index';
import configureStore from '../../../configureStore';
import history from '../../../utils/history';
import { render } from '@testing-library/react';

describe('<Footer />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, history);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      // tslint:disable-next-line: jsx-wrap-multiline
      <Provider store={store}>
        <IntlProvider locale="en">
          <Footer />
        </IntlProvider>
      </Provider>,
    );

    expect(firstChild).toMatchSnapshot();
  });
});
