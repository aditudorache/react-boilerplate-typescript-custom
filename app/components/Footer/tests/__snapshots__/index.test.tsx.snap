// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render and match the snapshot 1`] = `
<footer
  class="sc-bxivhb dtkJYr"
>
  <section>
    <span>
      This project is licensed under the MIT license.
    </span>
  </section>
  <section>
    <div
      class="sc-htpNat ffBkYG"
    >
      <select
        class="sc-bwzfXH eFNPLc"
      >
        <option
          value="en"
        >
          en
        </option>
        <option
          value="de"
        >
          de
        </option>
      </select>
    </div>
  </section>
  <section>
    <span>
      
      Made with love by 
      <a
        class="sc-bdVaJa cpswpp"
        href="https://twitter.com/mxstbr"
      >
        Max Stoiber
      </a>
      .
    
    </span>
  </section>
</footer>
`;
